org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/XMLServerLauncher.java

```diff
@@ -23,6 +26,8 @@
 
 import org.eclipse.lemminx.commons.ParentProcessWatcher;
 import org.eclipse.lemminx.customservice.XMLLanguageClientAPI;
+import org.eclipse.lemminx.io.NoOpInputStream;

```

Comment:
NoOpInputStream, NoOpInputStream, NoOpPrintStream are just used for XMLServerLauncher. I think it should be better to avoid exposing those classes which are not an API for contributions.

That's why I suggest you that you define NoOpInputStream, NoOpInputStream, NoOpPrintStream as static private class inside XMLServerLauncher.

Reply:
I'd rather keep them as separate classes. Keeping them public is no big deal, but if you prefer, I can move them to the lemminx package and keep them package visible only

Reply:
> I can move them to the lemminx package and keep them package visible only

Yes please. I would like to avoid that contributor uses those classes.